docs/configuration.md

```diff
@@ -0,0 +1,723 @@
+           "reference": "Binary/xxx",
+           "identifier": { "value": "patient_register" }
+         }
+       },
+       {
+         "title": "Patient Register configuration",
+         "mode": "working",
+         "focus": {
+           "reference": "Binary/xxx",
+           "identifier": { "value": "data_filters" }
+         }
+       },
+       {
+         "title": "Forms configuration",
+         "mode": "working",
+         "focus": {
+           "reference": "Binary/xxx",
+           "identifier": { "value": "forms" }
+         }
+       },
+       {
+         "title": "Sync configuration",
+         "mode": "working",
+         "focus": {
+           "reference": "Binary/xxx",
+           "identifier": { "value": "sync" }
+         }
+       }
+     ]
+   }
+   ```
+   
+2. Application config
+   
+   ```json
+   {
+     "appId": "xxx",
+     "classification": "application",
+     "theme": "AppTheme",
+     "languages": ["en", "sw"],
+     "applicationName": "App Name",
+     "appLogoIconResourceFile": "ic_default_logo",
+     "count": "100"
+   }
+   ```
+   
+3. Login config
+
+   ```json
+   {
+     "appId": "xxx",
+     "classification": "login",
+     "applicationName": "App Name",
+     "applicationVersion": "0.0.1",
+     "darkMode": false,
+     "showLogo": true,
+     "enablePin": true
+   }
+   ```
+
+4. App Feature config
+
+   ```json
+   {
+     "appId": "xxx",
+     "classification": "app_feature",
+     "appFeatures": [
+       {
+         "feature": "HouseholdManagement",
+         "active": true,
+         "settings": {},
+         "target": "CHW",
+         "healthModule": "FAMILY",
+         "useCases": [
+           "HOUSEHOLD_REGISTRATION"
+         ]
+       }
+     ]
+   }
+   ```
+
+5. Pin config
+
+   ```json
+   {
+     "appId": "xxx",
+     "classification": "pin",
+     "applicationName": "App Name",
+     "appLogoIconResourceFile": "ic_default_logo",
+     "showLogo": true
+   }
+   ```
+
+6. Data Filters config
+
+   ```json
+   {
+     "appId": "xxx",
+     "classification": "data_filters",
+     "filters": [
+       {
+         "id": "forms_list",
+         "filterType": "TOKEN",
+         "key": "context",
+         "valueType": "CODING",
+         "valueCoding": {
+           "system": "http://fhir.ona.io",
+           "code": "000002"
+         }
+       }
+     ]
+   }
+   ```
+
+7. Forms config
+
+   ```json
+   {
+     "appId": "xxx",
+     "classification": "forms",
+     "forms": [
+       {
+         "form": "family-registration",
+         "title": "Add Family",
+         "identifier": "82952",
+         "saveButtonText": "ADD FAMILY",
+         "setPractitionerDetails" : true,
+         "setOrganizationDetails" : true
+       }
+     ]
+   }
+   ```
+
+8. Patient Register config
+
+   ```json
+   {
+     "appId": "xxx",
+     "classification": "patient_register",
+     "newClientButtonText": "ADD FAMILY",
+     "registrationForm": "family-registration"
+   }
+   ```
+
+9. Sync config
+
```

Comment:
Thanks for the feedback.
Sure I'll do that

Reply:
By this complex expression `focus.subject eq Organization/#organization`
Can I assume the API request link will look like this?

https://fhir.labs.smartregister.org/fhir/[SOME_RESOURCE]?focus.subject=Organization/101

Correct me if I'm wrong.

Reply:
No. The tables looks good and example is also fine. Just wanted to mention that there could be numerous way someone would want to sync their data. And since its FHIR way of we need to point to FHIR docs on how to write a sync config expression or code i.e. how do some know that code can be patient.name or any other value. And also the expression can be simple or complex, so how do someone learn to write complex expressions. So If we can also add these FHIR references and explanation on possible values in sync config that would be very helpful.

The complete expression is "patient.organization eq #organization or focus.subject eq Organization/#organization".
So link might look like 
[https://fhir.labs.smartregister.org/fhir/Observation?_filter=patient.organization eq 361 or focus.subject eq Organization/361](https://fhir.labs.smartregister.org/fhir/Observation?_filter=patient.organization%20eq%20361%20or%20focus.subject%20eq%20Organization/361)

Reply:
I want to understand more about this case of complex expression:
`patient.organization eq #organization or focus.subject eq Organization/#organization`

`patient` is a SearchParameter from Observer.
It filters based on the path of `Observer.subject` it gets the subject inside the Observer.

Then how about the `patient.organization`?
Does that means it filters based on the Observer's Patient subject via the `patient` SearchParameter,
Then,
It gets the Patient's Organization via the `organization` SearchParameter? which Patient resource has.
https://build.fhir.org/patient.html#search

Is that right?

Reply:
yes. this is called chained parameters. https://hl7.org/fhir/search.html#chaining